compiler-core/src/build/project_compiler.rs

```diff
@@ -484,7 +500,12 @@ where
         };
         let config_path = package_root.join("gleam.toml");
         let config = PackageConfig::read(config_path, &self.io)?;
-        self.compile_gleam_package(&config, false, package_root)
+
+        // Since we're compiling a dependency package we always want to compile
+        // all its modules and check that it's compiling.
+        // That's why we're not using the project compiler's `compilation_scope`
+        // but we force the `AllModules` compilation scope.
+        self.compile_gleam_package(&config, false, package_root, CompilationScope::AllModules)
```

Comment:
One slight thing here: The second argument indicates whether the package is the root, so it knows enough to know whether it should access `self.compilation_scope` or not.

Further due to there now being two different arguments saying whether the package is the root it's possible for them to disagree and result in a logic bug. If `is_root` is false it should be possible to get to the path where it doesn't compile modules.

In `compile_gleam_package` I'd do something like this:

```rust
        let compile_modules = match compilation_scope {
            CompilationScope::OnlyDependencyModules if is_root => CompileModules::None,
            _ => CompileModules::All,
        };
```